extensions/sql-migration/src/wizard/loginMigrationStatusPage.ts

```diff
@@ -339,87 +344,114 @@ export class LoginMigrationStatusPage extends MigrationWizardPage {
 			return false;
 		}
 
-		await this._migrationProgressDetails.updateProperties({
-			'value': constants.MIGRATING_SERVER_ROLES_AND_SET_PERMISSIONS
-		});
-
-		result = await this.migrationStateModel._loginMigrationModel.MigrateServerRolesAndSetPermissions(this.migrationStateModel);
+		await this._updateMigrationProgressDetails(constants.ESTABLISHING_USER_MAPPINGS);
+		result = await this.migrationStateModel._loginMigrationModel.EstablishUserMappings(this.migrationStateModel);
 
 		if (!result) {
-			await this._migrationProgressDetails.updateProperties({
-				'value': constants.MIGRATING_SERVER_ROLES_AND_SET_PERMISSIONS_FAILED
-			});
-
+			await this._updateMigrationProgressDetails(constants.ESTABLISHING_USER_MAPPINGS_FAILED);
 			return false;
 		}
 
-		await this._migrationProgressDetails.updateProperties({
-			'CSSStyles': { 'display': 'none' },
-		});
+		await this._updateMigrationProgressDetails(constants.MIGRATING_SERVER_ROLES_AND_SET_PERMISSIONS);
+		result = await this.migrationStateModel._loginMigrationModel.MigrateServerRolesAndSetPermissions(this.migrationStateModel);
 
-		if (this.migrationStateModel._targetServerInstance) {
-			await this._migrationProgress.updateProperties({
-				'text': constants.LOGIN_MIGRATIONS_COMPLETED_STATUS_PAGE_DESCRIPTION(this._getTotalNumberOfLogins(), this.migrationStateModel.GetTargetType(), this.migrationStateModel._targetServerInstance.name),
-				'style': 'success',
-			});
-		} else {
-			await this._migrationProgress.updateProperties({
-				'text': constants.LOGIN_MIGRATIONS_COMPLETE,
-				'style': 'success',
-			});
+		if (!result) {
+			await this._updateMigrationProgressDetails(constants.MIGRATING_SERVER_ROLES_AND_SET_PERMISSIONS_FAILED);
+			return false;
 		}
 
-		this._progressLoader.loading = false;
-
-		this.wizard.doneButton.enabled = true;
-		return result;
+		await this._markMigrationSuccess();
+		return true;
 	}
 
 	private async _showLoginDetailsDialog(loginName: string): Promise<void> {
 		this.wizard.message = { text: '' };
 		const dialog = new MultiStepStatusDialog(
 			() => { });
 
-		const loginResults = this.migrationStateModel._loginMigrationModel.GetLoginMigrationResults(loginName);
+		const loginResults = this.migrationStateModel._loginMigrationModel.GetDisplayResults(loginName);
 		const isMigrationComplete = this.migrationStateModel._loginMigrationModel.isMigrationComplete;
 
 		await dialog.openDialog(constants.LOGIN_MIGRATIONS_LOGIN_STATUS_DETAILS_TITLE(loginName), loginResults, isMigrationComplete);
 	}
+
+	private _logMigrationStart(): void {
+		sendSqlMigrationActionEvent(
+			TelemetryViews.LoginMigrationStatusPage,
+			TelemetryAction.LoginMigrationStarted,
+			{
+				...getTelemetryProps(this.migrationStateModel),
+				'loginsAuthType': this.migrationStateModel._loginMigrationModel.loginsAuthType,
+				'numberLogins': JSON.stringify(this.migrationStateModel._loginMigrationModel.loginsForMigration.length),
```

Comment:
You can just call toString() - although generally you should be putting number values in measures instead

(same for all the other number properties you have in this PR)

Reply:
Sounds good. I will move it to measures. I followed the pattern in sql-migration is to put everything in properties.